x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_lifecycle/common/chart_components.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Axis, BarSeries, Chart, ScaleType, Settings } from '@elastic/charts';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLoadingChart,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { useElasticChartsTheme } from '@kbn/charts-theme';
+import type { TimeState } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import type { IlmPolicyPhases, PhaseName, Streams } from '@kbn/streams-schema';
+import { capitalize } from 'lodash';
+import moment from 'moment';
+import React, { useMemo } from 'react';
+import { orderIlmPhases } from '../helpers/helpers';
+import { formatBytes } from '../helpers/format_bytes';
+import type { DataStreamStats } from '../hooks/use_data_stream_stats';
+import { useIlmPhasesColorAndDescription } from '../hooks/use_ilm_phases_color_and_description';
+import { useIngestionRate, useIngestionRatePerTier } from '../hooks/use_ingestion_rate';
+
+interface ChartComponentProps {
+  definition: Streams.ingest.all.GetResponse;
+  stats?: DataStreamStats;
+  timeState: TimeState;
+  isLoadingStats: boolean;
+}
+
+export function ChartBarSeries({
```

Comment:
Do you foresee using this chart in other places? from what I see is coupled to ingestion so I'd not place it inside the `common` folder